src/main/java/org/kizombadev/markdownparser/entities/Syntax.java

```diff
@@ -1,22 +1,90 @@
+/*
+ * Marcel Swoboda
+ * Copyright (C) 2017
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
 package org.kizombadev.markdownparser.entities;
 
 
 import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.kizombadev.markdownparser.entities.interfaces.ImmutableSyntax;
+import org.kizombadev.markdownparser.entities.interfaces.MutableSyntax;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
-public abstract class Syntax {
-    private List<Syntax> children = new ArrayList<>();
+public class Syntax implements MutableSyntax {
+    private final List<ImmutableSyntax> children = new ArrayList<>();
+    private String content = null;
+    private SyntaxType type;
+
+    @NotNull
+    public static Syntax createWithChildren(SyntaxType type, ImmutableSyntax... children) {
+        checkNotNull(type);
+
+        Syntax syntax = new Syntax();
+        if (children != null) {
+            Arrays.stream(children).forEach(s -> syntax.addChild(s));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)